### PR TITLE
Wait for channel during broker test

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- minimal refactor to enable/disable tests easier
- update dependencies to kyma 1.9
- a channel is required during app broker test. Application creation cannot work without installing the required components anymore (SkipInstallation in Application-CR)
- wait for channel to be ready 
- split into two test-binaries (commandline arguments for tests do not really work with go 1.13)

**Related issue(s)**

### CODE REVIEW

File: tests/end-to-end/backup/pkg/common/common.go [hunk 1]
```diff
@@ -99,53 +96,60 @@ func TestBackupAndRestoreCluster(t *testing.T) {
...
 				err := myBackupClient.CreateNamespace(e2eTest.namespace)
-				So(err, ShouldBeNil)
+				convey.So(err, convey.ShouldBeNil)
 				t.Logf("[CreateResources: %s] Starting execution", e2eTest.name)
 				e2eTest.backupTest.CreateResources(e2eTest.namespace)
-				t.Logf("[CreateResources: %s] End with success", e2eTest.name)
-			}
-			for _, e2eTest := range e2eTests {
-				t.Logf("[TestResources: %s] Starting execution", e2eTest.name)
+				logrus.Infof("Testing resources in namespace: %s", e2eTest.namespace)
```
Comment (suleymanakbas91) [inline tests/end-to-end/backup/pkg/common/common.go]: Printing the test name would be better than printing the namespace.

File: tests/end-to-end/backup/test-cmd/before-backup/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM curlimages/curl:7.66.0
+COPY before-backup.test /
+ENTRYPOINT ["/before-backup.test"]
```
Comment (suleymanakbas91) [inline tests/end-to-end/backup/test-cmd/before-backup/Dockerfile]: Please add label to be consistent with the others and also copy the license folder as done in the previous dockerfile.  Please do the same for the other dockerfile as well.

File: tests/end-to-end/backup/pkg/common/common.go (hunk 1 above)
Comment (k15r) [reply tests/end-to-end/backup/pkg/common/common.go]: oh, i missed that

File: tests/end-to-end/backup/test-cmd/before-backup/Dockerfile (hunk 2 above)
Comment (k15r) [reply tests/end-to-end/backup/test-cmd/before-backup/Dockerfile]: sure

File: tests/end-to-end/backup/pkg/tests/servicecatalog/app_broker.go [hunk 3]
```diff
@@ -211,6 +230,58 @@ func (f *appBrokerFlow) createApplication() error {
...
+	}
+
+	_, err := f.messagingInterface.Channels(integrationNamespace).Create(channel)
+	if errors.IsAlreadyExists(err) {
+		return nil
+	}
+	return err
+}
+
+func (f *appBrokerFlow) waitForChannel() error {
```
Comment (anishj0shi) [inline tests/end-to-end/backup/pkg/tests/servicecatalog/app_broker.go]: why are we waiting for messagingchannel but not the httpsource?

File: tests/end-to-end/backup/pkg/tests/servicecatalog/app_broker.go [hunk 4]
```diff
@@ -211,6 +230,58 @@ func (f *appBrokerFlow) createApplication() error {
 	return err
 }
 
+// createChannel creates a Knative channel that would normally be installed by the application controller upon helm-chart installation. As we skip the installation part,
+// we need to ensure that the channel exists as it is required by the knative broker.
+func (f *appBrokerFlow) createChannel() error {
```
Comment (anishj0shi) [inline tests/end-to-end/backup/pkg/tests/servicecatalog/app_broker.go]: for an app broker does it makes sense to create channel? shouldn't this be done by the httpsource, right?

File: tests/end-to-end/backup/pkg/tests/servicecatalog/app_broker.go (hunk 3 above)
Comment (k15r) [reply tests/end-to-end/backup/pkg/tests/servicecatalog/app_broker.go]: In this case we do not send an event. so the only thing we are waiting for should be the channel.

**Pull request merged**